src/internal/ReplaySubject.ts

```diff
@@ -80,6 +96,15 @@ export class ReplaySubject<T> extends Subject<T> {
       subscriber.next(copy[i] as T);
     }
 
+    // Emit values that are synchronously pushed
+    for (let i = 0; i < syncValues.length && !subscriber.closed; i++) {
+      subscriber.next(syncValues[i]);
+    }
```

Comment:
maybe we should unshift emitted values to keep the syncValues small?
```ts
while(syncValues.length > 0 && !subscriber.closed) {
  subscriber.next(syncValues.shift());
}
```

Reply:
I'm pretty sure that shifting values in an array is less efficient than letting an array grow. The array will be discarded after the initial subscription is made.